model-optimizer/mo/front_ng/pipeline.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import logging as log
+from typing import List
+
+from mo.front_ng.extractor import fe_user_data_repack
+from mo.middle.passes.infer import validate_batch_in_shape
+
+from ngraph import Dimension, PartialShape        # pylint: disable=no-name-in-module,import-error
+from ngraph.frontend import Place                 # pylint: disable=no-name-in-module,import-error
+from ngraph.utils.types import get_element_type   # pylint: disable=no-name-in-module,import-error
+
+
+def moc_pipeline(argv: argparse.Namespace):
+    """
+    Load input model and convert it to nGraph function
+    :param: parsed command line arguments
+    :return: converted nGraph function ready for serialization
+    """
+    fem = argv.feManager
+    log.debug('Available front ends: {}'.format(
+        str(fem.get_available_front_ends())))
+    log.debug('Initializing new FE for framework {}'.format(argv.framework))
+    fe = fem.load_by_framework(argv.framework)
+    input_model = fe.load_from_file(argv.input_model)
+
+    user_shapes, outputs, freeze_placeholder = fe_user_data_repack(
+        input_model, argv.placeholder_shapes, argv.placeholder_data_types,
+        argv.output, argv.freeze_placeholder_with_value)
+
+    def check_places(places_original: List[Place], places_new: List[Place]):
+        """
+        Check if set of new places is same as original or not.
+        :param places_original: List[Place] Original model places
+        :param places_new: List[Place] New list of places
+        :return: True if new list of places is same as original
+        """
+        eq = len(places_original) == len(places_new)
+        if eq:
+            for item in places_original:
+                found = [x for x in places_new if x['node'].is_equal(item)]
+                if not found:
+                    eq = False
+                    break
+        return eq
```

Comment:
```python
return len(places_original) == len(places_new) and len([item in places_original if any([item.is_equal(item2) for item2 in places_new])]) == len(places_origin)
```

Reply:
Thanks, this also worked with couple corrections